dashboard/test/models/blockly_test.rb

```diff
@@ -412,6 +429,52 @@ def assert_equal_xml(a, b)
     assert_not_equal localized_custom_block, translated_block
   end
 
+  test 'option that contains a period in the key is translated' do
+    test_locale = :"fr-ST"
+    I18n.locale = test_locale
+    custom_i18n = {
+      "data" => {
+        "blocks" => {
+          "DanceLab_atSelectQuantity" => {
```

Comment:
I'd recommend not using an existing block, but instead making up a block definition just for this test